### ISSUE TITLE
Training results and metrics too ideal

### ISSUE THREAD

**Issue by masterbluster** [ultralytics/yolov5#6415]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

**I'm not sure if my results of training are reliable. All metrics seems to be too ideal:** 
![yolo_metrics](https://user-images.githubusercontent.com/39212570/150887211-813d9112-c535-40e6-9063-5333d3edd0df.png)

```
> wandb:             best/epoch 199
> wandb:           best/mAP_0.5 0.99443
> wandb:      best/mAP_0.5:0.95 0.93278
> wandb:         best/precision 0.98526
> wandb:            best/recall 0.99085
> wandb:        metrics/mAP_0.5 0.99443
> wandb:   metrics/mAP_0.5:0.95 0.93277
> wandb:      metrics/precision 0.98527
> wandb:         metrics/recall 0.99085
```
After using `val.py` all predictions are at 1.0 confidence:
![val_batch1_pred](https://user-images.githubusercontent.com/39212570/150888119-6372ad7a-d054-4269-ae1c-b915bca14cee.jpg)

I am trying to train pretrained yolov5s.pt on my custom dataset annotated in Roboflow consisting only one class - Pipe.
Dateset contains ~12000 images with ~18000 annotations and ~800 background images,  resized to 640x640 and  all split into training, validation and test sets.

I've trained with below command:
`python train.py --img 640 --batch 32 --epochs 200 --data /path/to/dataset/data.yaml  --weights yolov5s.pt --cache`

Is it possible to achive that good results? Or am I missing something?

**Comment by glenn-jocher** [comment]
@masterbluster if worse results is what you're after you could always try some other architectures.

**Comment by pasmai** [comment]
Is this the only explanation?

I have a dataset of roughly 4k images with 8 different classes. They all have different positions, rotations as well as lighting conditions. Yet, after training the model recognises everything always with 100% confidence perfectly.

Is it simply not challenged enough? Even for the small v5s and v5n models.

**Comment by glenn-jocher** [comment]
If you've split your dataset correctly and trained then your results are your results. 

![Screenshot 2021-08-04 at 13 15 05](https://user-images.githubusercontent.com/26833433/152557102-fced3e41-858f-4ac5-bb44-307c82d7ef75.jpg)

**Comment by MuShanZihahaha** [comment]
Hello, could you tell me whether the txt file of the background image is set to empty, or does not need the corresponding txt file?

**Comment by glenn-jocher** [comment]
@MuShanZihahaha both options work correctly for background images.

**Comment by MuShanZihahaha** [comment]
Excuse me, I am in the coco128 dataset, in order to make some pictures as background pictures, I deleted 24 txt files in labels, and two txt files are empty, which appeared during my training: 104 found, 24 missing, 2 empty, 0 corrupted, is 24 pictures without txt files considered as background pictures, or 2 pictures with empty txt as background pictures?

**Comment by MuShanZihahaha** [comment]
I am using yolov5:5.0

**Comment by glenn-jocher** [comment]
both
